bears/installation_tool.py

```diff
@@ -0,0 +1,161 @@
+import argparse
+import importlib
+import re
+import subprocess
+import sys
+
+from coala_utils.string_processing.StringConverter import StringConverter
+from coala_utils.Question import ask_question
+from coalib.collecting.Collectors import get_all_bears_names
+from coalib.collecting.Importers import iimport_objects
+from coalib.misc.Shell import call_without_output
+from prompt_toolkit import prompt
+from prompt_toolkit.contrib.completers import WordCompleter
+from pygments.lexers import SqlLexer
+from pyprint.ConsolePrinter import ConsolePrinter
+
+
+def install_pip_package(package_name):
+    """
+    Uses call to install a PyPi package.
+
+    :param package_name: The package to be installed.
+    """
+    print('To install the dependencies correctly, we will need administrator '
+          'rights from you.')
+    subprocess.call(['sudo', sys.executable, '-m',
+                     'pip', 'install', package_name, '--upgrade'])
+
+
+def get_output(command):
+    """
+    Runs the command and decodes the output and returns it.
+
+    :param command: The command to be run.
+    :return:        The output of the command, decoded.
+    """
+    function = subprocess.Popen(command, stdout=subprocess.PIPE)
+    return function.communicate()[0].decode("utf-8")
+
+
+def check_failed_bears(bear_list):
+    """
+    Checks if any bear failed installation. Exits with 1 if so.
+
+    :param bear_list: List of the failed bears.
+    """
+    if bear_list:
+        print('Bears that failed installing:' + "\n".join(bear_list))
+        sys.exit(1)
+
+
+def install_requirements(package_name):
+    """
+    Imports a package and tries installing its requirements.
+
+    :param package_name:        The package to be imported.
+    :param package_failed_list: The list with the packages which their
+                                requirements failed installing.
+    :return:                    A list with the packages which had their
+                                requirements failing to be installed.
+    """
+    package_failed_list = []
+    package_object = importlib.import_module(
+        package_name + "." + package_name)
+    for requirement in getattr(package_object, package_name).REQUIREMENTS:
+        try:
+            subprocess.call([requirement.install_command()])
```

Comment:
Um, this already returns a list? https://github.com/coala-analyzer/coala/blob/fee67b3089626b6b4f2e57e2ef1c6b6f335b7802/coalib/bears/requirements/PipRequirement.py#L36